### PR TITLE
patch/update flow with multiple entities

### PR DESCRIPTION
**Pull request by ereteog**
> **Epic** #https://github.com/advthreat/iroh/issues/2249

This PR refactors the patch and update flow to accept a list of patches/entities in perspective of bulk update/patch.
The ids are no more passed as parameters but provided in the patches/entities.

<a name="qa">[§](#qa)</a> QA
============================

This PR is a non negligeable refactoring, check that the CRUD update and patch operations works as expected.

### CODE REVIEW

File: src/ctia/flows/crud.clj [hunk 1]
```diff
@@ -75,34 +74,37 @@
...
       (id/str->short-id id))))
 
-(defn gen-random-uuid []
-  (UUID/randomUUID))
-
-(defn make-id
-  [entity-type]
-  (str (name entity-type) "-" (gen-random-uuid)))
-
-(s/defn ^:private find-entity-id :- s/Str
```
Comment (ereteog) [inline src/ctia/flows/crud.clj:85]: This function used to implicitely consider the type of flow from its content. `(find-id prev-entity)` was only related to update and patch and the other elements to different create cases.
I decomposed it in `find-create-entity-id` and `find-existing-entity-id`.

**Review by gbuisson — APPROVED** [review]
LGTM for the general code part,
for the rest I rely on the tests and my faith in functional programming.

File: src/ctia/entity/casebook.clj [hunk 2]
```diff
@@ -124,150 +123,97 @@
...
+        (PATCH "/" []
+          :body [partial-casebook PartialNewCasebook {:description "a Casebook partial update"}]
+          :summary "Partially Update a Casebook"
+          (if-let [res (patch-flow identity
+                                   identity-map
+                                   wait_for
+                                   :replace
+                                   partial-casebook
+                                   id)]
+            (ok res)
```
[the remark this answers is not in the document]
Comment (ereteog) [reply src/ctia/entity/casebook.clj:176]: It's done it in the `patch-flow` fn above which factorize the code. Then in the other cases it's now done twice, I need to clean it :-)

**Review by yogsototh — APPROVED** [review]
EDIT wrong PR :D haha

File: test/ctia/flows/crud_test.clj [hunk 3]
```diff
@@ -192,11 +252,15 @@
...
-                      {"sighting-1" true "sighting-2" true})
+                      {sighting-id-1 true sighting-id-2 true})
          (delete-flow "delete-flow must not create events for deleted entities"
-                      {"sighting-1" false "sighting-2" false})
+                      {sighting-id-1 false sighting-id-2 false})
          (delete-flow "delete flow deletes entities and creates events only for existing entities when some are not found"
-                      {"sighting-3" true "missing1" false "missing2" false})
-         ;; TODO test concurrent deletes
+                      {sighting-id-3 true "missing1" false "missing2" false})
          )))))
```
Comment (agzam) [inline test/ctia/flows/crud_test.clj:266]: dangling parens

**Review by agzam — APPROVED** [review]

**Pull request merged**